docs/Displaying a User Interface/Popup Keyboards/index.md

```diff
@@ -177,10 +179,58 @@ KeyboardListener keyboardListener = new KeyboardListener() {
     public void onKeyboardDidSendEvent(KeyboardEvent event, String currentInputText) {
         // This is sent upon every event, such as keypresses, cancellations, and aborting
     }
+
+    @Override
+    public void onKeyboardDidUpdateInputMask(KeyboardEvent event) {
+        switch (event) {
+            case INPUT_KEY_MASK_ENABLED:
+                // The user enabled input key masking
+                break;
+            case INPUT_KEY_MASK_DISABLED:
+                // The user disabled input key masking
+                break;
+            default:
+                break;
+        }
+    }
 };
 ```
 !@
 
+@![iOS, android, javaSE, javaEE]
+### Configuring the Keyboard Properties
```

Comment:
You didn't add anything to describe custom keys (add an image), the numeric layout (add an image), describing what "mask input characters" is, or about how to check keyboard layout capabilities.

We should probably add considerably more detail to this section about each potential keyboard configuration option.

Reply:
Agree. I can add more details.

Reply:
@joeljfischer I updated the PR to add more details as suggested.